client/express-checkout/blocks/components/express-checkout-component.js

```diff
@@ -29,6 +29,26 @@ const getPaymentMethodsOverride = ( enabledPaymentMethod ) => {
 	};
 };
 
+// Visual adjustments to horizontally align the buttons.
+const adjustButtonHeights = ( buttonOptions, expressPaymentMethod ) => {
+	// Apple Pay has a nearly imperceptible height difference. We increase it by 1px here.
+	if ( buttonOptions.buttonTheme.applePay === 'black' ) {
+		if ( expressPaymentMethod === 'applePay' ) {
+			buttonOptions.buttonHeight = buttonOptions.buttonHeight + 0.4;
+		}
```

Comment:
@asumaran This breaks the ECE implementation when you're using the Largest size of button because `buttonHeight` is at `55` which is already the maximum allowed value to be passed to the ECE API. We need to fix this because none of the WooPayments buttons render on the max size, including the WooPay button:

<img width="378" alt="image" src="https://github.com/user-attachments/assets/95a9a6ef-aadc-47ab-9d34-04ea34090099">

Reply:
Anytime a merchant has ECE enabled and is using the `Large` size of the buttons. Probably not very often? Although I honestly can't say, it's hard to measure AFAIK.

Reply:
I've submitted a PR fixing this. https://github.com/Automattic/woocommerce-payments/pull/9327

Reply:
>Anytime a merchant has ECE enabled and is using the Large size of the buttons. 

@pierorocca, apart from that, it only happens on the block-based cart and checkout pages. It's working as usual on the product page, shortcode-based cart, and checkout pages.